src/qofonosimwatcher.cpp

```diff
@@ -150,7 +161,8 @@ QOfonoSimWatcher::QOfonoSimWatcher(QObject *parent) :
 
 QOfonoSimWatcher::~QOfonoSimWatcher()
 {
-    // d_ptr is a child object, it gets deleted automatically
```

Comment:
The problem goes the other way around - object is being destructed when someone is processing its event, i.e. by the signal handler.